src/components/AxisCollection/index.js

```diff
@@ -1,72 +1,121 @@
-    yAxisWidth: PropTypes.number,
-    yTransformations: PropTypes.objectOf(
-      PropTypes.shape({
-        k: PropTypes.number.isRequired,
-        y: PropTypes.number.isRequired,
-        rescaleY: PropTypes.func.isRequired,
-      })
-    ).isRequired,
-  };
-
-  static defaultProps = {
-    series: [],
-    zoomable: true,
-    updateYTransformation: () => {},
-    yAxisWidth: 50,
-  };
+const propTypes = {
+  height: PropTypes.number.isRequired,
+  width: PropTypes.number.isRequired,
+  series: seriesPropType,
+  zoomable: PropTypes.bool,
+  updateYTransformation: PropTypes.func,
+  yAxisWidth: PropTypes.number,
+  yTransformations: PropTypes.objectOf(
+    PropTypes.shape({
+      k: PropTypes.number.isRequired,
+      y: PropTypes.number.isRequired,
+      rescaleY: PropTypes.func.isRequired,
+    })
+  ).isRequired,
+  axisDisplayMode: axisDisplayModeType,
+  onMouseEnter: PropTypes.func,
+  onMouseLeave: PropTypes.func,
+};
 
-  shouldComponentUpdate() {
-    // TODO: Implement
-    return true;
-  }
+const defaultProps = {
+  series: [],
+  zoomable: true,
+  updateYTransformation: () => {},
+  yAxisWidth: 50,
+  axisDisplayMode: AxisDisplayMode.ALL,
+  onMouseEnter: null,
+  onMouseLeave: null,
+};
 
-  render() {
+class AxisCollection extends React.Component {
+  renderAllAxes() {
     const {
+      axisDisplayMode,
       series,
       zoomable,
-      yAxisWidth,
       height,
-      width,
       updateYTransformation,
+      yAxisWidth,
       yTransformations,
     } = this.props;
-    let axisOffsetX = 0;
+    // Make sure that the axes don't appear if we're also going to be rendering
+    // the placeholder axis.
+    let axisOffsetX =
+      axisDisplayMode === AxisDisplayMode.COLLAPSED ? yAxisWidth : 0;
+    return series.filter(s => !s.hidden).map((s, idx) => {
+      if (idx > 0) {
+        axisOffsetX += yAxisWidth;
+      }
+      return (
+        <YAxis
+          key={`y-axis--${s.id}`}
+          offsetx={axisOffsetX}
+          zoomable={s.zoomable !== undefined ? s.zoomable : zoomable}
+          series={s}
+          height={height}
+          width={yAxisWidth}
+          updateYTransformation={updateYTransformation}
+          yTransformation={yTransformations[s.id]}
+        />
+      );
+    });
+  }
+
+  renderPlaceholderAxis() {
+    const { axisDisplayMode, height, yAxisWidth } = this.props;
+    if (axisDisplayMode !== AxisDisplayMode.COLLAPSED) {
+      return null;
+    }
+    return (
+      <CollapsedAxis
+        key="y-axis--collapsed"
+        height={height}
+        width={yAxisWidth}
+      />
+    );
+  }
+
+  render() {
+    const {
+      width,
+      height,
+      axisDisplayMode,
+      onMouseEnter,
+      onMouseLeave,
+    } = this.props;
+    const axes = [];
+    if (axisDisplayMode === AxisDisplayMode.COLLAPSED) {
+      axes.push(this.renderPlaceholderAxis());
+    }
+    // We need to render all of the axes (even if they're hidden) in order to
```

Comment:
This points to another problem about how state is kept in the axis, i'll file an issue to remember this.